crates/optimism/txpool/src/validator.rs

```diff
@@ -147,11 +147,42 @@ where
 
         let outcome = self.inner.validate_one(origin, transaction);
 
+        self.apply_op_checks(outcome)
+    }
+
+    /// Validates all given transactions.
+    ///
+    /// Returns all outcomes for the given transactions in the same order.
+    ///
+    /// See also [`Self::validate_one`]
+    pub fn validate_all(
+        &self,
+        transactions: Vec<(TransactionOrigin, Tx)>,
+    ) -> Vec<TransactionValidationOutcome<Tx>> {
+        transactions
+            .into_iter()
+            .map(|(origin, tx)| {
+                if tx.is_eip4844() {
+                    TransactionValidationOutcome::Invalid(
+                        tx,
+                        InvalidTransactionError::TxTypeNotSupported.into(),
+                    )
+                } else {
+                    let outcome = self.inner.validate_one(origin, tx);
+                    self.apply_op_checks(outcome)
+                }
+            })
+            .collect()
```

Comment:
this is just the same as validate_one ?

so unclear why this is now inlined

Reply:
`validate_one_with_provider` isn't exposed on `EthTransactionValidator` yet, see https://github.com/paradigmxyz/reth/issues/15298
the optimisation introduced here is bailing early in case of eip4844 tx

Reply:
unsure what this means in this context.

I'm fine with extracting the op checks, but we can just use validate_one here